packages/dynamoose/lib/Model/index.ts

```diff
@@ -439,21 +439,21 @@ export class Model<T extends ItemCarrier = AnyItem> extends InternalPropertiesCl
 	// latestTableDetails: DynamoDB.DescribeTableOutput;
 	// pendingTaskPromise: () => Promise<void>;
 	Item: typeof ItemCarrier;
```

Comment:
@fishcharlie Is being able to access the underlying `Item` class from a model part of public interface? If it is not, then there aren't any issues. However if it is, then we'll have to figure out a way to better type this property.

For example, this is possible:

```js
const UserModel = dynamoose.model('User', userSchema)
const user = new UserModel.Item({ id: 'foobar' })
```

This works fine in JS, however typing this correctly in TS has proven challenging. I made a naive attempt at fixing the types: 

```suggestion
	Item: T & typeof ItemCarrier;
```

However, TypeScript understandably isn't happy with this, as the internal `Item` class created within `Model` is not assignable to the generic we pass in:

```
Type 'typeof Item' is not assignable to type 'T & typeof Item'.
  Type 'typeof Item' is not assignable to type 'T'.
    'T' could be instantiated with an arbitrary type which could be unrelated to 'typeof Item'.
```

Haven't been able to find a good solution to this yet.

Reply:
@tranhl 

> Is being able to access the underlying Item class from a model part of public interface?

It might work. But it's not supported. If someone tries to do that in their code I would point them to this: https://dynamoosejs.com/other/FAQ#can-i-use-an-undocumented-property-class-method-or-function-in-dynamoose.

Reply:
Okay perfect 👍 .

It's also mentioned in the TS docs that [documentation should be treated as the source of truth](https://dynamoosejs.com/getting_started/TypeScript#should-i-trust-the-documentation-or-typescript-typings-more), so we can set this aside until `Model.Item` becomes public interface.